core/src/main/java/fj/data/Enumerator.java

```diff
@@ -185,6 +181,18 @@ private Enumerator(final F<A, Option<A>> successor, final F<A, Option<A>> predec
     return Stream.fromFunction(this, id, a);
   }
 
+  /**
+   * Returns a stream of the values from this enumerator,
+   * starting at the min of given Bounded, ending at the max, counting up.
+   *
+   * @param bounded A value at which to begin the stream.
+   * @return a stream of the values from this enumerator, cut by bounded, counting up.
+   */
+  public Stream<A> toStream(final Bounded<A> bounded) {
+    final F<A, A> id = identity();
+    return Stream.fromFunction(this, id, bounded.min()).takeWhile(item -> !item.equals(bounded.max()));
```

Comment:
takeWhile should be 
```suggestion
    return Stream.fromFunction(this, id, bounded.min()).takeWhile(item ->order.isLessThanOrEqualTo(item, bounded.max()));
```
otherwise you miss Bounded.max() in the Stream. (also we avoid the use `Object#equals`).